shotover/src/sources/mod.rs

```diff
@@ -11,10 +11,10 @@ pub mod cassandra;
 pub mod kafka;
 pub mod redis;
 
-#[derive(Deserialize, Debug, Copy, Clone)]
+#[derive(Deserialize, Debug, Clone)]
 pub enum Transport {
     Tcp,
-    WebSocket,
+    WebSocket(#[serde(skip)] String),
```

Comment:
Rather than smuggling the protocol name in via the Transport Config, why dont we have the codec specify what protocol it would use for websockets.
```rust
pub trait CodecBuilder: Clone + Send {
    type Decoder: DecoderHalf;
    type Encoder: EncoderHalf;
    fn build(&self) -> (Self::Decoder, Self::Encoder);

    fn new(direction: Direction) -> Self;
    
    fn websocket_subprotocol(&self) -> &'static str;
}
```